### DISCUSSION TITLE
How to get all setting?

### DISCUSSION THREAD

**Discussion by freehzw** [rawilk/laravel-settings#36] [Q&A] [answered]
Now get a setting value must by key name, I'd like to get all setting values one time. Thank you

---

When set encryption false, but the key still encrypted. I wish the `key` column not encrypted.

**Comment by rawilk** [comment]
This is not currently possible as is with the package. Currently the package uses `Rawilk\Settings\Support\KeyGenerator` to generate the key used for the setting, which is generating an md5 of the key combined with a serialized Context object. There currently isn't a way to override this functionality but I may add it in later. I would recommend forking your own version if you need this functinality.

**Comment by rawilk** [comment, marked as answer]
I've just releases v3 of this package, which adds support for both fetching multiple settings at once, and serializing your own keys for storage. I recommend reading the [Upgrade Guide](https://randallwilk.dev/docs/laravel-settings/v3/upgrade) to get started upgrading.
